### DISCUSSION TITLE
Automatically add -XstartOnFirstThread on Mac?

### DISCUSSION THREAD

**Discussion by Phillipus** [eclipse-tycho/tycho#234] [Q&A] [answered]
Hi,

I'm frequently adding the following to poms to support running tests on Mac using Surefire:

```
<profile>
    <id>osx</id>
    <activation>
        <os>
            <family>mac</family>
        </os>
    </activation>
    <build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.eclipse.tycho</groupId>
                    <artifactId>tycho-surefire-plugin</artifactId>
                    <version>${tycho-version}</version>
                    <configuration>
                        <argLine>-XstartOnFirstThread</argLine>
                    </configuration>
                </plugin>
            </plugins>
        </pluginManagement>
    </build>
</profile>
```

I also found this old bug discussing automatically adding it:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=388084

But I'm not clear whether this was addressed or not, or if there's some other way to use the -XstartOnFirstThread argument when running Maven/Tycho without having to add the profile above?

Thanks in advance,

Phil

**Comment by mickaelistria** [comment, marked as answer]
> But I'm not clear whether this was addressed or not

It was closed as WONTFIX and no related issue mentions any related development, and the issue is still here and the profile is still necessary, so it looks like it was not addressed.

> if there's some other way to use the -XstartOnFirstThread argument when running Maven/Tycho without having to add the profile above?

Not until tycho-surefire-plugin receives a patch that automatically pass adds flag for mac ;)

**Comment by Phillipus** [comment]
Thanks for the answer.

> Not until tycho-surefire-plugin receives a patch that automatically pass adds flag for mac ;)

Perhaps I will look into it.

**Comment by vogella** [comment]
Opened https://github.com/eclipse/tycho/discussions/234
